guacamole/src/main/java/org/apache/guacamole/rest/sharingprofile/APISharingProfile.java

```diff
@@ -0,0 +1,207 @@
+     *     The sharing profile to use to populate the data of the new
+     *     APISharingProfile.
+     */
+    public APISharingProfile(SharingProfile sharingProfile) {
+
+        // Set main information
+        this.name = sharingProfile.getName();
+        this.identifier = sharingProfile.getIdentifier();
+        this.primaryConnectionIdentifier = sharingProfile.getPrimaryConnectionIdentifier();
+        
+        // Associate any attributes
+        this.attributes = sharingProfile.getAttributes();
+
+    }
+
+    /**
+     * Returns the human-readable name of this sharing profile.
+     *
+     * @return
+     *     The human-readable name of this sharing profile.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Set the human-readable name of this sharing profile.
+     *
+     * @param name
+     *     The human-readable name of this sharing profile.
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+    
+    /**
+     * Returns the unique string which identifies this sharing profile within
+     * its containing directory.
+     *
+     * @return
+     *     The unique string which identifies this sharing profile within its
+     *     containing directory.
+     */
+    public String getIdentifier() {
+        return identifier;
+    }
+
+    /**
+     * Sets the unique string which identifies this sharing profile within
+     * its containing directory.
+     *
+     * @param identifier
+     *     The unique string which identifies this sharing profile within its
+     *     containing directory.
+     */
+    public void setIdentifier(String identifier) {
+        this.identifier = identifier;
+    }
+    
+    /**
+     * Returns the identifier of the primary connection that this sharing
+     * profile can be used to share.
+     *
+     * @return
+     *     The identifier of the primary connection that this sharing profile
+     *     can be used to share.
+     */
+    public String getPrimaryConnectionIdentifier() {
+        return primaryConnectionIdentifier;
+    }
+
+    /**
+     * Sets the identifier of the primary connection that this sharing profile
+     * can be used to share.
+     *
+     * @param primaryConnectionIdentifier
+     *     The identifier of the primary connection that this sharing profile
+     *     can be used to share.
+     */
+    public void setPrimaryConnectionIdentifier(String primaryConnectionIdentifier) {
+        this.primaryConnectionIdentifier = primaryConnectionIdentifier;
+    }
+
+    /**
+     * Returns a map of all associated connection parameter values which apply
+     * when the sharing profile is used, indexed by parameter name.
+     *
+     * @return
+     *     A map of all associated connection parameter values which apply when
+     *     the sharing profile is used, indexed by parameter name.
+     */
+    public Map<String, String> getParameters() {
+        return parameters;
```

Comment:
Yup. Don't want to expose parameters unless explicitly needed, as they usually contain sensitive information. Like with `APIConnection`, the parameters of `APISharingProfile` aren't ever really populated in the outbound direction; they're used for update based on received data from the client. If the parameters are needed by the client, that information is queried directly from the `.../parameters` resource, and permission to do *that* is explicitly denied unless `UPDATE` permission is available (as it's an information leak otherwise).